### PR TITLE
Finalize release notes for 3.2

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Finalize 3.2 release notes
- add closed issues and known bugs and drop irrelevant entries
- manual fixup pass
- drop "working draft" from the title
- add 3.2.0 from the release list

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Moved 50737 from known issue to resolved.

**Review by gmarull — DISMISSED** [review]

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -1713,7 +1713,6 @@ Known Issues
 - :github:`41822` - BLE IPSP sample cannot handle large ICMPv6 Echo Request
```
Comment (stephanosio) [inline doc/releases/release-notes-3.2.rst:1674]: a few additional issues to drop from the list:

```
- :github:`50596` - Documentation: Broken links in the previous release documentation
- :github:`50844` - zcbor module apis which are used for mcu boot functionality are not building in cpp file against v3.1.0
```

**Review by stephanosio — COMMENTED** [review]
also some additional fixes:

```
s/CONFIG_ALWAYS_SWITCH_THROUGH_ECALL/CONFIG_RISCV_ALWAYS_SWITCH_THROUGH_ECALL/g
```

**Comment by fabiobaltieri** [comment]
Updated: fixed CONFIG_ALWAYS_SWITCH_THROUGH_ECALL, dropped 50596 and 50844, added 50861 and 50306 that got fixed in the meantime.

**Review by stephanosio — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**